packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -696,13 +730,302 @@ describe("Garbage Collection Tests", () => {
+					baseSnapshot,
+					gcBlobMap,
+					currentGCVersion,
+				);
+
+				// GC state, tombstone state and deleted nodes should all be read from base snapshot.
+				const baseSnapshotData = await garbageCollector.baseSnapshotDataP;
+				assert(
+					baseSnapshotData !== undefined,
+					"base snapshot was not initialized correctly",
+				);
+				assert(
+					baseSnapshotData.gcState !== undefined,
+					"GC state in base snapshot should not be available",
+				);
+				assert(
+					baseSnapshotData.tombstones !== undefined,
+					"Tombstone state in base snapshot should be available",
+				);
+				assert(
+					baseSnapshotData.deletedNodes !== undefined,
+					"Deleted nodes in base snapshot should be available",
+				);
+
+				// Initialize from the base state and validate that tombstones and deleted state both have one entry
+				// as per the base snapshot.
+				await garbageCollector.initializeBaseState();
+				assert.strictEqual(garbageCollector.tombstones.length, 1);
+				assert.strictEqual(garbageCollector.deletedNodes.size, 1);
+			});
+
+			it("discards GC state and tombstone state in base snapshot when GC version changes", async () => {
+				const updatedGCVersion = currentGCVersion + 1;
+				const { baseSnapshot, gcBlobMap } = getSnapshotWithGCVersion(updatedGCVersion);
+				const garbageCollector = createGCOverride(
+					baseSnapshot,
+					gcBlobMap,
+					updatedGCVersion,
+				);
+
+				// GC state and tombstone state should be discarded but deleted nodes should be read from base snapshot.
+				const baseSnapshotData = await garbageCollector.baseSnapshotDataP;
+				assert(
+					baseSnapshotData !== undefined,
+					"base snapshot was not initialized correctly",
+				);
+				assert(
+					baseSnapshotData.gcState === undefined,
+					"GC state in base snapshot should be undefined when GC version changes",
+				);
+				assert(
+					baseSnapshotData.tombstones === undefined,
+					"Tombstone state in base snapshot should be undefined when GC version changes",
+				);
+				assert(
+					baseSnapshotData.deletedNodes !== undefined,
+					"Deleted nodes in base snapshot should be available",
+				);
+
+				// Initialize from the base state and validate that tombstones has 0 entry because it was discarded.
+				// Deleted nodes should have one entry because it is still used.
+				await garbageCollector.initializeBaseState();
+				assert.strictEqual(garbageCollector.tombstones.length, 0);
+				assert.strictEqual(garbageCollector.deletedNodes.size, 1);
+			});
+
+			it("reads all GC data from when refreshing from snapshot with same GC version", async () => {
+				const { baseSnapshot, gcBlobMap } = getSnapshotWithGCVersion(currentGCVersion);
+				const garbageCollector = createGCOverride(
+					baseSnapshot,
+					gcBlobMap,
+					currentGCVersion,
+				);
+
+				// GC state, tombstone state and deleted nodes should all be read from base snapshot.
+				const baseSnapshotData = await garbageCollector.baseSnapshotDataP;
+				assert(
```

Comment:
I wouldn't bother asserting these since it's redundant with the other test. Or maybe pile them into one assert with message "PRECONDITION FAILED" or something.

I wish Mocha had a `precondition` module with `precondition.assert` functions to match `assert` functions but if they failed, it doesn't fail the test the same as if a regular assert fails.  Would help reduce investigation surface area when something fundamental breaks a million tests, ya know?